### PR TITLE
Transmission Compressor unit conversions

### PR DESCRIPTION
**Pull request by mbarlow12**
The `TransmissionCompressor` was raising errors when activated:
```
pint.errors.DimensionalityError: Cannot convert from 'kilopascal ** 0.333333 * mile ** 0.333333 / kilometer ** 0.333333 / psi_absolute ** 0.333333' ([mass] ** 5.55112e-17 / [time] ** 1.11022e-16) to 'fraction' (dimensionless)
```
Stack trace:
```
opgee/processes/transmission_compressor.py:80: in run
    Compressor.get_compressor_energy_consumption(
opgee/processes/compressor.py:130: in get_compressor_energy_consumption
    total_work, outlet_temp, outlet_press = Compressor.get_compressor_work_temp(field,
opgee/processes/compressor.py:43: in get_compressor_work_temp
    z_factor = gas.Z_factor(inlet_reduced_temp, inlet_reduced_press)
opgee/thermodynamics.py:1006: in Z_factor
    pr = reduced_pressure.to("frac").m
../../../../anaconda3/envs/opgee/lib/python3.11/site-packages/pint/facets/plain/quantity.py:536: in to
    magnitude = self._convert_magnitude_not_inplace(other, *contexts, **ctx_kwargs)
../../../../anaconda3/envs/opgee/lib/python3.11/site-packages/pint/facets/plain/quantity.py:480: in _convert_magnitude_not_inplace
    return self._REGISTRY.convert(self._magnitude, self._units, other)
../../../../anaconda3/envs/opgee/lib/python3.11/site-packages/pint/facets/plain/registry.py:1050: in convert
    return self._convert(value, src, dst, inplace)
../../../../anaconda3/envs/opgee/lib/python3.11/site-packages/pint/facets/context/registry.py:405: in _convert
    return super()._convert(value, src, dst, inplace)
../../../../anaconda3/envs/opgee/lib/python3.11/site-packages/pint/facets/nonmultiplicative/registry.py:259: in _convert
    return super()._convert(value, src, dst, inplace)
```
To address this, the following updates are included:
- add test and necessary fixtures for the `TransmissionCompressor` process
- add unit conversions to get ratio values into `fraction` units

Other updates:
- misc import ordering and autoformatting
- update `HeavyOilUpgrading` output stream name

### CODE REVIEW

**Comment by mbarlow12** [comment]
@rjplevin @wennanlong I made a VERY rough test case here where I basically copied values found while running the gas lifting field. If possible, please review the assertions and let me know if/how they need to change. Thanks!

File: tests/files/test_model.xml [hunk 1]
```diff
@@ -894,6 +895,37 @@
...
+			<Contains>gas</Contains>
+			<Component name="CO2" phase="gas">119.35794</Component>
+			<Component name="H2" phase="gas">18.04195</Component>
+			<Component name="H2S" phase="gas">18.48612</Component>
+			<Component name="C1" phase="gas">108.77169</Component>
+			<Component name="C2" phase="gas">122.32514</Component>
+			<Component name="C3" phase="gas">119.59151</Component>
+			<Component name="C4" phase="gas">78.81646</Component>
+			<A name="temperature">60</A>
+			<A name="pressure">14.676</A>
```
Comment (mbarlow12) [inline tests/files/test_model.xml:912]: Note that these values come from peeking at the input stream contents when running the `gas_lifting_field` example.

File: tests/test_processes.py [hunk 2]
```diff
@@ -350,7 +351,15 @@ def test_GasReinjectionCompressor(test_model):
...
-
+    
+def test_TransmissionCompressor(test_model):
+    analysis = test_model.get_analysis('test_gas_processes')
+    field = analysis.get_field('test_TransmissionCompressor')
+    field.run(analysis)
+    proc = field.find_process("TransmissionCompressor")
+    total = proc.energy.data.sum()
+    expected = ureg.Quantity(1733.3619, "mmbtu/day")
+    assert approx_equal(total, expected, rel=10e-3)
```
Comment (mbarlow12) [inline tests/test_processes.py:362]: The expected quantity here is copied from the return of `proc.energy.data.sum()`. I have no idea if these values are appropriate.

File: opgee/process.py
Comment (rjplevin) [inline opgee/process.py:1]: I see more removals of local variables. I hope you're not doing this everywhere in the code base, as I would oppose it. Third reason: it's easier to see values in the debugger. (In pycharm, variables are directly visible, while one has to click on `self` to see its instance variables. BTW, if you're not using an IDE with good debugger, switch to PyCharm.

**Review by rjplevin — CHANGES_REQUESTED** [review]
I went through the whole thing twice. Apart from my comments, it looks fine.

I would prefer to revert the removal of local variables, but it's not critical. However, I would request that, unless you can convince me of the benefit, you don't do more of that elsewhere in the code.

**Pull request closed without merging**